pkg/cmd/pipeline/start_test.go

```diff
@@ -1898,665 +685,1655 @@ func Test_start_pipeline_imageRes_withExistingRes_createNew(t *testing.T) {
+					return nil
+				},
+			},
+		},
+		{
+			name:               "Create new cloud resource with existing resource",
+			namespace:          "ns",
+			input:              cs11,
+			last:               false,
+			serviceAccountName: "svc1",
+			serviceAccounts:    []string{"task1=svc1"},
+			prompt: prompt.Prompt{
+				CmdArgs: []string{"cloudpipeline"},
+				Procedure: func(c *expect.Console) error {
+					if _, err := c.ExpectString("Choose the cloudEvent resource to use for cloudres"); err != nil {
+						return err
+					}
+
+					if _, err := c.Send(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("create new \"cloudEvent\" resource"); err != nil {
+						return err
+					}
+
+					if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("Enter a name for a pipeline resource :"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine("newcloudresource"); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("Enter a value for targetURI :"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine("https://10.10.10.10"); err != nil {
+						return err
+					}
+
+					if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectEOF(); err != nil {
+						return err
+					}
+
+					tekton := cs11.Pipeline.Tekton()
+					runs, err := tekton.PipelineRuns("ns").List(v1.ListOptions{})
+					if err != nil {
+						return err
+					}
+
+					if runs.Items != nil && runs.Items[0].Spec.PipelineRef.Name != "cloudpipeline" {
+						return errors.New("pipelinerun not found")
+					}
+
+					c.Close()
+					return nil
+				},
 			},
 		},
 	}
-	opts := startOpts("ns", cs, false, "svc1", []string{"task1=svc1"})
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			opts.RunPromptTest(t, test)
+	for _, tp := range testParams {
+		t.Run(tp.name, func(t *testing.T) {
+			p := test.Params{
+				Kube:   tp.input.Kube,
+				Tekton: tp.input.Pipeline,
+			}
+			p.SetNamespace(tp.namespace)
+
+			opts := startOptions{
+				cliparams:          &p,
+				Last:               tp.last,
+				ServiceAccountName: tp.serviceAccountName,
+				ServiceAccounts:    tp.serviceAccounts,
+			}
+
+			tp.prompt.RunTest(t, tp.prompt.Procedure, func(stdio terminal.Stdio) error {
+				opts.askOpts = prompt.WithStdio(stdio)
+				opts.stream = &cli.Stream{Out: stdio.Out, Err: stdio.Err}
+
+				return opts.run(tp.prompt.CmdArgs[0])
+			})
 		})
 	}
 }
 
-func Test_start_pipeline_imageRes_withExistingRes_useExisting(t *testing.T) {
-
-	pipelineName := "imagepipeline"
-
-	cs, _ := test.SeedTestData(t, pipelinetest.Data{
-		Pipelines: []*v1alpha1.Pipeline{
-			tb.Pipeline(pipelineName, "ns",
-				tb.PipelineSpec(
-					tb.PipelineDeclaredResource("imageres", "image"),
-				),
-			),
-		},
+func Test_start_pipeline(t *testing.T) {
```

Comment:
The test cases below from here are doing its own checks.
So it seems to be hard integrating into same table driven test cases and there is no merit doing it, I left these as these.